server/requestorserver/server.go

```diff
@@ -555,13 +591,13 @@ func (s *Server) checkAuth(w http.ResponseWriter, r *http.Request, rerr *irma.Re
 	}
 	if !applies {
 		var ctype = r.Header.Get("Content-Type")
-		if ctype != "application/json" && ctype != "text/plain" {
+		if !regexp.MustCompile("^application/json").MatchString(ctype) && !regexp.MustCompile("^text/plain").MatchString(ctype) {
```

Comment:
Is this double negation correctly? If I'm right it says 'Not (No match for application/json in ctype)'. It seems the opposite from the previous code. What was the problem with that code after all?